base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -940,13 +940,18 @@ pub fn calculate_mmr_roots<T: BlockchainBackend>(db: &T, block: &Block) -> Resul
     }
 
     for input in body.inputs().iter() {
-        let index = db
-            .fetch_mmr_leaf_index(MmrTree::Utxo, &input.output_hash())?
-            .ok_or_else(|| ChainStorageError::ValueNotFound {
-                entity: "UTXO".to_string(),
-                field: "hash".to_string(),
+        // db search should be much faster and should be the most widely used case,
+        // there for we search first in the db, then in the new mmr
```

Comment:
nit: Rather explain the logic than guessing about perf

```suggestion
        // Search the DB for the output leaf index so that it can be marked as spent/deleted.
        // If the output hash is not found, check the current output_mmr. This allows zero-conf transactions
```